src/cache/client/npm/test/test.js

```diff
@@ -270,6 +270,26 @@ describe('vulnRegexDetector', () => {
 					const response3 = vulnRegexDetector.testSync(pattern, invalidConfig);
 					assert.ok(response3 === vulnRegexDetector.responses.invalid, `Query succeeded? response ${response3}`);
 				});
+				it('should not return an expired cache value', () => {
+					const pattern = 'abcde'; // TODO perhaps make some temporary cache for all of the tests and clear it each time.
+					// Make sync query to prime local persistent cache, but use negative cache value to ensure expiration.
+					let validConfig = { cache: { type: vulnRegexDetector.cacheTypes.persistent, expirationTime: -1 } };
+					const response1 = vulnRegexDetector.testSync(pattern, validConfig);
+					assert.ok(response1 === vulnRegexDetector.responses.safe, `Error, unexpected response for sync query: ${response1}`);
+
+					let invalidConfig = {
+						server: {
+							hostname: 'no such host',
+							port: 1
+						},
+						cache: {
+							type: vulnRegexDetector.cacheTypes.persistent,
+							expirationTime: -1
```

Comment:
Suggest you declare a `cacheConfig` variable at the top of the function and use it here and above.